joomla/joomla-cms
# [com_fields] Move some events from system plugin to content plugin

Pull Request for Issue -.

### Summary of Changes
Plugin events are triggered by the dispatcher in order specified by the plugins administration component. However some plugin types like system plugins are dispatched before content plugins.

The pull requests moves _onContentAfterTitle_, _onContentBeforeDisplay_ and _onContentAfterDisplay_ from fields system plugin to fields content plugin - so it's possible to define where output will be rendered on the page.

I've prepared an example where I'm using both vote plugin and fields plugin to show a gallery.  Both are configured to show on _onContentBeforeDisplay_ event.
The goal is to 

### Testing Instructions
1) Switch On the vote plugin
  _Extensions > Plugins > Content - vote_, make sure the _Position_ is set to _Top_
2) Add new Custom field
  _Content > Fields > New_, make sure the _Automatic Display_ is set to _Before Display_
3) Change ordering o plugins 
_Extensions > Plugins >  Search Tools_
Select Type: Content, Sort table by: Ordering Ascending

### Expected result
User can choose the order of content plugins to display
![j-fields-after](https://cloud.githubusercontent.com/assets/612953/24287908/c0bf022c-107b-11e7-9283-a3a384134d34.png)

### Actual result
The fields plugin output is always displayed before any other content plugins.
![j-fields-before](https://cloud.githubusercontent.com/assets/612953/24287922/c690828e-107b-11e7-81b5-14f2bef45ca2.png)

### Documentation Changes Required
none

cc @laoneo
